src/components/reviews/addReviewForm.jsx

```diff
@@ -6,36 +6,126 @@ class AddReviewForm extends React.Component {
     super(props);
 
     this.state = {
-
+      recommend: null,
+      characteristics: {},
+      summary: "",
+      body: "",
+      photos: [],
+      nickname: "",
+      email: "",
+      valid: true
     };
+    this.onChangeBound = this.onChange.bind(this);
+    this.changeCharacteristicBound = this.changeCharacteristic.bind(this);
+    this.submitBound = this.submit.bind(this);
+  }
 
+  onChange(event) {
+    const name = event.target.name;
+    const value = event.target.value;
+    this.setState({
+      [name]: value
+    });
   }
 
+  changeCharacteristic(event) {
+    const characteristic = event.target.name;
+    const value = event.target.value;
+    this.setState({
+      characteristics: {...this.state.characteristics, [characteristic]: value}
+    });
+  }
 
+  submit(event) {
+    event.preventDefault();
+  }
 
   render() {
     return (
       <div>
         <div className="subtitle">{`About the ${this.props.name}`}</div>
         <form>
           <label>
-            Overall rating*
+            <div className="add-input">Overall rating*</div>
             [rating selector goes here]
           </label>
           <br />
           <label>
-            Do you recommend this product?*
+          <div className="add-input">Do you recommend this product?*</div>
             <div>
-              <input type="radio" name="recommend" value="true" /> Yes
-              <input type="radio" name="recommend" value="false" /> No
+              <input type="radio" name="recommend" value={true} onChange={this.onChangeBound} /> Yes
+              <input type="radio" name="recommend" value={false} onChange={this.onChangeBound} /> No
             </div>
           </label>
           <br />
           <div>
             {Object.keys(this.props.characteristics).map((charName) => {
-              return <CharacteristicsRadio characteristic={charName} />;
+              return <CharacteristicsRadio key={charName} characteristic={charName} handler={this.changeCharacteristicBound} />;
             })}
           </div>
+          <label>
+          <div className="add-input">Summary:*</div>
+            <br />
+            <input
+              className="input" 
+              type="text" 
+              name="summary" 
+              value={this.state.summary} 
+              maxLength="60" 
+              required={true} 
+              onChange={this.onChangeBound} />
+          </label>
+          <label>
+          <div className="add-input">Your review:*</div>
+            <br />
+            <textarea 
```

Comment:
May pick your brain on how to implement text area. I am unfamiliar with using this input feature.